### PR TITLE
EXUI-1988 - refactor of reasonable adjustments find change code.

### PR DESCRIPTION
**Pull request by anthonydummer**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/EXUI-1983

### Change description ###
Refactor of the reasonable adjustments code to check for changes.  Have removed the logic that was checking against case flags and have introduced a check against the hmc and shv individualdetails reasonable adjustments. 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] tests have been updated / new tests has been added (if needed)
- [x] Does this PR introduce a breaking change

### CODE REVIEW

**Review by StoneBenHMCTS — APPROVED** [review]

**Review by StoneBenHMCTS — APPROVED** [review]
LGTM

File: src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.spec.ts [hunk 1]
```diff
@@ -1186,4 +1191,294 @@ describe('HearingEditSummaryComponent', () => {
     const expected = cold('(b|)', { b: toShow });
     expect(result$).toBeObservable(expected);
   });
+
+  // New tests for change to
```
Comment (andywilkinshmcts) [inline src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.spec.ts:1195]: Incomplete comment?

File: src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts [hunk 2]
```diff
@@ -444,49 +446,55 @@ export class HearingEditSummaryComponent extends RequestHearingPageFlow implemen
...
-      : '';
-
-    // Return true if there are changes in reasonable adjustments and/or language interpreter flags
-    if (flagIdsSHV.join() !== partyFlagIdsString) {
-      return true;
-    }
-    // There are no changes for reasonable adjustments and language interpreter flags when compared SHV with HMC
-    return false;
+    individualParties.forEach(
+      (party) => party.individualDetails?.reasonableAdjustments?.sort((a, b) => {
```
Comment (andywilkinshmcts) [inline src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts:493]: Is there a reason you are using a special sort function here? It looks pretty standard - you should be able to use reasonableAdjustments?.sort()

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Fix incomplete comment and consider if you need a special sort function for reasonable adjustments

File: src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.spec.ts (hunk 1 above)
Comment (anthonydummer) [reply src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.spec.ts:1195]: Should have removed this, it was for me to identify the new test I was creating.  This has been removed.

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**